fuzzer/src/main.rs

```diff
@@ -1,16 +1,45 @@
 use std::collections::{BTreeMap, BTreeSet, VecDeque, LinkedList, BinaryHeap};
 use std::time::Duration;
 
-use bitvec::{vec::BitVec, order::Msb0};
+use bitvec::{vec::BitVec, order::Msb0, bitvec, order::BitOrder, store::BitStore};
 use honggfuzz::fuzz;
 use parity_scale_codec::{Encode, Decode, Compact};
+use honggfuzz::arbitrary::{Arbitrary, Unstructured, Result as ArbResult};
 
-#[derive(Encode, Decode)]
+#[derive(Encode, Decode, Clone, PartialEq, Debug, Arbitrary)]
 pub struct MockStruct{
 	vec_u: Vec<u8>
 }
 
-#[derive(Encode, Decode)]
+#[derive(Encode, Decode, Clone, PartialEq, Debug)]
+/// Used for implementing the Arbitrary trait for a BitVec.
+pub struct BitVecWrapper<O: BitOrder, T: BitStore>(BitVec<O, T>);
+
+impl<O: 'static + BitOrder, T: 'static + BitStore + Arbitrary> Arbitrary for BitVecWrapper<O, T> {
+	fn arbitrary(u: &mut Unstructured<'_>) -> ArbResult<Self> {
+		let v = Vec::<T>::arbitrary(u)?;
+		Ok(BitVecWrapper(BitVec::<O, T>::from_vec(v)))
+	}
+}
+
+
+// #[derive(Encode, Decode, Debug, Clone, Arbitrary)]
+// pub struct BitcVecWrapper<BitVec>(BitVec);
```

Comment:
Should this commented code be removed?

Reply:
My bad code is not ready yet I'm switching back to draft mode :)